File: src/main/java/com/core/base/corebase/repository/AnnouncementRepository.java
package com.core.base.corebase.repository;

import com.core.base.corebase.domain.organization.Announcement;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface AnnouncementRepository extends JpaRepository<Announcement, Long> {
    List<Announcement> findByOrganizationIdOrderByCreatedAtDesc(Long organizationId);
}

File: src/main/kotlin/com/core/base/corebase/repository/AnnouncementRepository.kt
package com.core.base.corebase.repository

import com.core.base.corebase.domain.organization.Announcement
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface AnnouncementRepository : JpaRepository<Announcement, UUID> {
    fun findByOrganizationIdOrderByCreatedAtDesc(organizationId: UUID) : List<Announcement>
}